Sprint-3/revise/implement/is-prime.js

```diff
@@ -0,0 +1,36 @@
+function isPrime(num) {
+    //function will check whether the given number is a prime number according to the conditions given
+    let count = 0;
+    if (num >= 1) {
+
+        //divide the given number by the iterations if the number is divisible by (1 and itself ONLY) then it is prime and will have a count of 2
+        //Otherwise it is not a prime number
+        for (let i = 1; i <= num; i++) {
```

Comment:
This approach works and your code is correct, but the approach is not very efficient. Can you try to improve its performance? Since you may not be familiar with the mathematical properties of prime numbers, just use Google or ChatGPT to learn a better approach (not JS implementation) to determine if a number is prime, and then try to implement the approach.

Reply:
This is an opportunity to practice "reading code".
May I suggest this: Pick some numbers (e.g., 15, 17, 19, 21, 23) and trace the code to see how it works,
with or without using the Python visualizer, https://pythontutor.com/javascript.html#mode=edit

Have you tried asking GPT to explain the code for you?

There is also a simpler version that skips only even numbers. You can ask GPT to present you this simpler version and try to understand this simpler version first. It could help you understand the version you showed here.

Reply:
Hi. I've implemented the simpler version in my code that skips even numbers. I also took the version above and put it into a visualiser. Made it a lot easier to understand.  I prefer the version that skips even numbers, it's just a bit more straightforward